src/modules/src/planner.c

```diff
@@ -88,6 +88,16 @@ bool plan_is_stopped(struct planner *p)
 	return p->state == TRAJECTORY_STATE_IDLE;
 }
 
+void plan_disable(struct planner *p)
```

Comment:
So now that we have a plan_disable, is the plan_stop necessary in the grander scheme of things? We do have a s[topping function in the crtp_commander_generic](https://github.com/bitcraze/crazyflie-firmware/blob/56e357e0f4a3f01e04b0849214ef3628292f76d2/src/modules/src/crtp_commander_generic.c#L84). 

Isn't it better to put plan_disable() in the stop function of the generic commander, and use the more general stop command instead. Just to prevent the same functionality in multiple places.

Reply:
I believe we still need both the stopped and disabled states in the `planner.c` state machine due to reason 3 in my original PR comment. However, with this PR `crtpCommanderHighLevelStop` is no longer called by other parts of the firmware. The only place it might still be used is due to radio commands. I am not sure but it seems like we could deprecate and re-route the high-level stop radio command; I can't see why it needs to be different from the low-level one.

Reply:
Just mind that crtpCommanderHighLevelStop are being used in the the [app api examples](https://github.com/bitcraze/crazyflie-firmware/blob/master/app_api/src/app_main.c) and [the app implementation of the swarm_demo](https://github.com/bitcraze/crazyflie-firmware/blob/master/examples/demos/swarm_demo/app.c)

Reply:
so we just have to make sure to replace those. And the CI would catch it as well as we check all the app layer examples and api

Reply:
@knmcguire in the swarm demo example, I believe the `crtpCommanderHighLevelStop` is not needed...

It is used here after landing: https://github.com/bitcraze/crazyflie-firmware/blob/250a9e552d793895351898f75d1529d9c12ed148/examples/demos/swarm_demo/app.c#L395-L401 but this same check and state transition will happen automatically within `planner.c`:
https://github.com/bitcraze/crazyflie-firmware/blob/250a9e552d793895351898f75d1529d9c12ed148/src/modules/src/planner.c#L93-L97

(IIRC, this automatic state transition is the only reason we have a `land` command at all - otherwise we could just use `goTo` followed by a manual stop.)

It is used here when a crash is detected: https://github.com/bitcraze/crazyflie-firmware/blob/250a9e552d793895351898f75d1529d9c12ed148/examples/demos/swarm_demo/app.c#L425-L427 but the supervisor framework should be overriding the high-level commander anyway: https://github.com/bitcraze/crazyflie-firmware/blob/250a9e552d793895351898f75d1529d9c12ed148/src/modules/src/supervisor.c#L115-L122

I am not sure what's going on in the first link - it looks like it's just calling every API function in arbitrary order.

Reply:
perhaps @krichardsson could take a look at the demo example to make sure it's not broken when we remove it. And then if in the same PR we also remove the app_api one as well, then it should be all good for the checks

Reply:
I think we can remove it from the swarm demo.

Since crtpCommanderHighLevelStop() is part of the public API we should not remove it completely. I assume the correct way is to NOP it and make it deprecated. I think it should be kept in app_api.c for a while, whith a comment that it has been deprecated.

Reply:
Could we mark ut with __attribute__(deprecated)__?